### DISCUSSION TITLE
Edited navigation.html

### DISCUSSION THREAD

**Discussion by muzzug** [piharpi/jekyll-klise#72] [General]
Hi.

If there is anyone here who, like me, is bothered by the weird navigation behaviour for previous/next posts (on the first post "next" is on the left, on the others on the right and on the last one "previous" is on the left), I offer a modified version. Just use the code to replace the code in navigation.html

```
<nav class="post-nav">
  <div class="post-nav-item post-nav-prev">
    <div class="nav-arrow"> Previous </div>
    {% if page.previous %}
    <a href="{{ page.previous | relative_url }}">
      <span class="post-title">{{ page.previous.title }}</span>
    </a>
    {% endif %}
  </div>
    
  <div class="post-nav-item post-nav-next">
    <div class="nav-arrow">Next</div>
    {% if page.next %}
    <a href="{{ page.next | relative_url }}">
      <span class="post-title">{{ page.next.title }}</span>
    </a>
    {% endif %}
  </div> 
</nav>
```

Just beware - only the title of the previous/next article is active in this case. The "next" and "previous" itself no longer work as a link as in the original code. :)

**Comment by BrandonHargitay** [comment]
Hey!
I updated your code snippet to add class attributes to the navigation links for the previous and next pages, and includes conditional statements to display the links and page titles only if they exist. The purpose of these changes is to maintain the styling of the original navigation bar while improving the functionality of the links. I noticed your change didn't have the same styling as the OG nav.

```
<nav class="post-nav">
  <div class="post-nav-item post-nav-prev">
    <div class="nav-arrow"> Previous </div>
    {% if page.previous %}
    <a href="{{ page.previous | relative_url }}">
      <span class="post-title">{{ page.previous.title }}</span>
    </a>
    {% endif %}
  </div>
    
  <div class="post-nav-item post-nav-next">
    <div class="nav-arrow">Next</div>
    {% if page.next %}
    <a href="{{ page.next | relative_url }}">
      <span class="post-title">{{ page.next.title }}</span>
    </a>
    {% endif %}
  </div> 
</nav>
```

**Comment by piharpi** [comment]
Thanks @muzzug @BrandonHargitay, I would consider this discussion/suggestion in to next version.
